benchmark/util/text-decoder.js

```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+const common = require('../common.js');
+
+const bench = common.createBenchmark(main, {
+  encoding: ['utf-8', 'latin1', 'iso-8859-3'],
+  ignoreBOM: [0, 1],
+  len: [256, 1024 * 16, 1024 * 512],
+  n: [1e6]
+});
+
+function main({ encoding, len, n, ignoreBOM }) {
+  const buf = Buffer.allocUnsafe(len);
```

Comment:
heh, relying on the buffer being filled with uninitialized memory is fun ;-) ... This works for the sake of the test but it's unconventional